subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -179,20 +179,20 @@ struct ll_conn *ll_connected_get(u16_t handle)
 	return conn;
 }
 
-u8_t ull_conn_allowed_check(void *conn)
+u8_t ull_conn_allowed_check(void *conn) /* This function can be replaced by atomic_cas */
 {
 	struct ll_conn * const conn_hdr = conn;
-	if (conn_hdr->llcp_req != conn_hdr->llcp_ack) {
-		return BT_HCI_ERR_CMD_DISALLOWED;
-	}
 
-	conn_hdr->llcp_req++;
-	if (((conn_hdr->llcp_req - conn_hdr->llcp_ack) & 0x03) != 1) {
-		conn_hdr->llcp_req--;
-		return BT_HCI_ERR_CMD_DISALLOWED;
+	int ret = atomic_cas((atomic_t*)&conn_hdr->llcp_req,
+						 conn_hdr->llcp_ack,
+						 (conn_hdr->llcp_req + 1));
+
+	/* If ret == 1, llcp_req has been updated */
+	if (ret == 1) {
+		return 0;
 	}
 
-	return 0;
+	return BT_HCI_ERR_CMD_DISALLOWED;
```

Comment:
Thanks for your feedback. I can put this in a separate commit, but I don't see why this would be desired. The functionality should be unchanged from before this PR.

Reply:
Agree. Just a nitpick. But I know some people who do care. Now you're prepared ;)